evaluate.py

```diff
@@ -132,23 +130,38 @@ def run_test(init_op, dataset):
 
             bar.finish()
 
+            # Print test summary
             wer, cer, samples = calculate_report(wav_filenames, ground_truths, predictions, losses)
             mean_loss = np.mean(losses)
+            print('Test on %s - WER: %f, CER: %f, loss: %f' % (dataset, wer, cer, mean_loss))
+            print('-' * 80)
 
-            # Take only the first report_count items
-            report_samples = itertools.islice(samples, FLAGS.report_count)
+            # Take only the first, median and last report_count items
+            best_samples = samples[:FLAGS.report_count]
+            worst_samples = samples[-FLAGS.report_count:]
+            median_index = int(len(samples) / 1.5)
+            median_left = int(FLAGS.report_count / 2)
+            median_right = FLAGS.report_count - median_left
+            median_samples = samples[median_index - median_left:median_index + median_right]
 
-            print('Test on %s - WER: %f, CER: %f, loss: %f' %
-                  (dataset, wer, cer, mean_loss))
-            print('-' * 80)
-            for sample in report_samples:
-                print('WER: %f, CER: %f, loss: %f' %
-                      (sample.wer, sample.cer, sample.loss))
+            def print_single_sample(sample):
```

Comment:
I'd prefer if we had clearly the information in the output that this is `worst` / `median` / `best` samples

Reply:
Like this:

-----
Examples with best WER score:
----
... some examples ...
-----
Examples with median WER score:
----
... some examples ...
-----
Examples with worst WER score:
----
... some examples ...
----

?

Reply:
No need to make the output too heavy in text, but that's the idea, yeah. It could just be worded as `Worst WER:`, `Median WER:` and `Best WER:`